lib/src/utils/schema/getSection.ts

```diff
@@ -0,0 +1,26 @@
+import VernaJSONSchemaType from ':/types/rjsf';
+import { Maybe } from ':/types/utils';
+import { getSectionName } from ':/utils';
+
+export function getSection(
+  selector: Maybe<string>,
+  schema: VernaJSONSchemaType,
+  id: string,
+): VernaJSONSchemaType {
+  if (selector) {
+    if (!schema)
+      throw new Error("Error, getSection can't get the properties of the related section");
+    return schema.properties![selector as keyof typeof schema.properties];
+  } else {
+    const sectionName = getSectionName(id);
+    if (sectionName) {
+      if (!schema.properties?.[sectionName])
+        throw new Error("Error, getSection can't get the properties of the related section");
+      return schema.properties?.[sectionName];
+    } else {
+      if (!schema)
+        throw new Error("Error, getSection can't get the properties of the related section");
+      return schema;
+    }
+  }
```

Comment:
I think this block could be simplified if you flat the if statements

Reply:
Yes, I now simplified it as much as I could